src/detect-tcpmss.c

```diff
@@ -62,6 +62,33 @@ void DetectTcpmssRegister(void)
     return;
 }
 
+/**
+ * \brief Return human readable value for tcp.mss mode
+ *
+ * \param mode uint8_t DetectU16Data tcp.mss mode value
+ */
+const char *DetectTcpmssModeToString(DetectUintMode mode)
```

Comment:
This function should rather be in rust/src/detect/uint.rs as it is generic for integers, not specific to tcp mss, right ?

Reply:
Better still to be in the `detect-engine-uint.c` file, I think.

Reply:
detect-engine-uint.c is just a wrapper to rust/src/detect/uint.rs

You can use rust/src/detect/uint.rs directly ;-)

Reply:
Well, Idk.

I don't know rust, & I just got accustomed to C during this exercise.
So I made sure not to go too deep.

Reply:
Ok, in the end, I think we will need it to be rust

Reply:
> Ok, in the end, I think we will need it to be rust

Hello, so I was looking through the `uint.rs` source file for this, and I was thinking; "Wouldn't it be better if the function is placed in the `detect-engine-uint.c` file instead?".

Reply:
Really sorry for none of us ever getting to answer these before.

> Will I have to wrap the function into the detect-engine-uint.c file too?

Usually, if we need a Rust function to be available to the C side, we will do that still from the Rust code, (see for instance https://github.com/OISF/suricata/blob/master/rust/src/detect/uint.rs#L303)

> Based on my thoughts, I'd have to write a function that uses the return values of function detect_parse_uint_mode

I don't think that your function would change much from what it looks like today, just that it should be in the Rust side, as Philippe mentioned. You are already using the right Parameter type, and then would still return a string.
Comparing to `detect_parse_uint_mode`,  it looks to me that you would almost have params and return in reverse from what is there :P